bin/pipeline_result_processor/prp/parse.py

```diff
@@ -0,0 +1,338 @@
+#! /usr/bin/env python
+"""Combine output files from pipeline into a standardized json output."""
+
+import csv
+import json
+import logging
+from collections import defaultdict
+from typing import Any, Dict, Tuple
+from unittest import result
+
+import click
+import pandas as pd
+
+from .models.metadata import SoupVersion, SoupVersions
+from .models.phenotype import (
+    PhenotypeResult,
+    PhenotypeType,
+    ResistanceGene,
+    ResistanceVariant,
+    VirulenceGene,
+)
+from .models.sample import AssemblyQc, MethodIndex
+from .models.typing import TypingMethod, TypingResultCgMlst, TypingResultMlst
+
+LOG = logging.getLogger(__name__)
+
+
+SPP_MIN_READ_FRAC = 0.001
+
+
+def parse_qust_results(file: str) -> AssemblyQc:
+    """Parse quast file and extract relevant metrics.
+
+    Args:
+        sep (str): seperator
+
+    Returns:
+        AssemblyQc: list of key-value pairs
+    """
+    LOG.info(f"Parsing tsv file: {file}")
+    creader = csv.reader(file, delimiter="\t")
+    header = next(creader)
+    raw = [dict(zip(header, row)) for row in creader]
+    return AssemblyQc(
+        total_length=int(raw[0]["Total length"]),
+        reference_length=raw[0]["Reference length"],
+        largest_contig=raw[0]["Largest contig"],
+        n_contigs=raw[0]["# contigs"],
+        n50=raw[0]["N50"],
+        assembly_gc=raw[0]["GC (%)"],
+        reference_gc=raw[0]["Reference GC (%)"],
+        duplication_ratio=raw[0]["Duplication ratio"],
+    )
+
+
+def parse_mlst_results(path: str) -> TypingResultMlst:
+    """Parse mlst results from mlst to json object."""
+    LOG.info("Parsing mlst results")
+    result = json.load(path)[0]
+    result_obj = TypingResultMlst(
+        scheme=result["scheme"],
+        sequence_type=None
+        if result["sequence_type"] == "-"
```

Comment:
Use `if not` and then remove the `else` line?

Reply:
Does this indentation work? Just make sure it works, I do not follow the syntax with `if` statements lacking `:` at the end of the row, and then no indentation for the row starting with `alleles`

Reply:
Im not sure the `else` statement cant be removed in an assignment of a variable

Reply:
ah, now I see the parentheses I did not see before